extensions/objects/models/objects.py

```diff
@@ -1633,208 +1552,265 @@ def get_schema(cls):
+            dict. The object schema.
+        """
+        if cls._value_key_name is None or cls._value_schema is None:
+            raise NotImplementedError(
+                'The _value_key_name and _value_schema for this class must '
+                'both be set.')
+        return {
+            'type': 'dict',
+            'properties': [{
+                'name': 'contentId',
+                # The default content id is none. However, it should be
+                # populated before being saved. The normalize() method has
+                # validation checks for this.
+                'schema': {'type': 'unicode_or_none'}
+            }, {
+                'name': cls._value_key_name,
+                'schema': copy.deepcopy(cls._value_schema),
+            }]
+        }
+
+    @classmethod
+    def normalize(cls, raw):
+        """Validates and normalizes a raw Python object.
+
+        Args:
+            raw: dict. A Python object to be validated against the schema,
+                normalizing if necessary.
+
+        Returns:
+            dict. The normalized object.
+
+        Raises:
+            TypeError. Error while normalizing.
+        """
+        if not isinstance(raw['contentId'], python_utils.BASESTRING):
+            raise TypeError(
+                'Expected content id to be a string, received %s' %
+                raw['contentId'])
+
+        return {
+            'contentId': raw['contentId'],
+            cls._value_key_name: schema_utils.normalize_against_schema(
+                cls.normalize_value(raw[cls._value_key_name]),
+                cls._value_schema)
+        }
+
+
+class TranslatableUnicodeString(BaseTranslatableObject):
     """Class for translatable unicode strings."""
 
     _value_key_name = 'unicodeStr'
     _value_schema = UnicodeString.get_schema()
-
     default_value = {
         'contentId': None,
-        'unicodeStr': ''
+        'unicodeStr': '',
     }
 
     @staticmethod
-    def _normalize_value(raw):
-        """Validates and normalizes the value fields of the translatable object.
+    def normalize_value(value):
+        """Normalizes the translatable value of the object.
 
         Args:
-            raw: *. A translatable Python object whose values are to be
-                normalized.
+            value: *. The translatable part of the Python object (corresponding
+                to the non-content-id field) which is to be normalized.
 
         Returns:
-            dict. A normalized translatable Python object with its values
-            normalized.
-
-        Raises:
-            TypeError. The Python object cannot be normalized.
+            str. The normalized value.
         """
-        if not isinstance(raw['unicodeStr'], python_utils.BASESTRING):
-            raise TypeError(
-                'Invalid unicode string: %s' % raw['unicodeStr'])
+        if not isinstance(value, python_utils.BASESTRING):
+            raise TypeError('Invalid unicode string: %s' % value)
```

Comment:
If the schema is `UnicodeString.get_schema()` then the default normalize method will do these checks, right?

Reply:
Simplified as above -- thanks!